### DISCUSSION TITLE
RelationManager Table HeaderActions refresh - Filament V3

### DISCUSSION THREAD

**Discussion by clinvest** [filamentphp/filament#9579] [Help] [answered]
Hi all,

It's been hours I am struggling with this situation, not finding any info.

I use Filament V3.

I'm trying to conditionally show or hide a headerAction onmy relationManager table. Here is my code:

```
Action::make('send_estimate')
      ->label(__('admin.send_estimate'))
      ->color('green-3')
      ->outlined()
      ->visible(function (RelationManager $livewire): bool { var_dump($livewire->getOwnerRecord()->management_status->value); return true; })
      ->action(function ($record): void {
          $record->save();
      })
```
The point is, when I update my ownerRecord from a relation $record, it is not live updating my headerActions.

Means, I need to refresh the browser to get my button visible or hidden after action is submited.

I don't want to use the poll for performance reason (right?)

How can I implement a listener on a RelationManager table headerAction and refresh them?

Thank you in advance for your help.

**Comment by leandrocfe** [comment]
You can use [livewire events](https://livewire.laravel.com/docs/events)

```php
->visible(fn (RelationManager $livewire): bool => $livewire->getOwnerRecord()->management_status->value)
->action(function ($livewire): void {
    $record = $livewire->getOwnerRecord();
    $record->management_status->value = false;
    $record->save();
})
->after(fn ($livewire) => $livewire->dispatch('update-record'))
```

EditPage.php
```php
#[On('update-record')]
public function updateRecord()
{
    $this->refreshFormData([
        'management_status',
        //...
    ]); 
}
```

**Reply by leandrocfe** [reply, marked as answer]
[the opposite](https://github.com/filamentphp/filament/discussions/8510#discussioncomment-7018501)

**Reply by clinvest** [reply]
thank you very much lendro the opposite did it for me I needed the list watch.
